### ISSUE TITLE
Incorrect lint error that Binding is never used

### ISSUE THREAD

**Issue by c-git** [intellij-rust/intellij-rust#10253]
## Environment

* **IntelliJ Rust plugin version:** 0.4.190.5263-223
* **Rust toolchain version:** 1.67.0 (fc594f156 2023-01-24) x86_64-unknown-linux-gnu
* **IDE name and version:** PyCharm 2022.3.3 Professional Edition (PY-223.8836.43)
* **Operating system:** Linux 5.19.0-35-generic
* **Macro expansion:** enabled

## Problem description
Getting incorrect linting warning for bindings never used on variables that are used inline in format strings.

## Steps to reproduce
I'm not sure what the general steps to reproduce are but the following code sample the minimal one that I could produce. I get a warning in PyCharm that "e" is unused however it is used in the panic statement. Same behaviour also observed in match statements.

```rust
fn main() {
    if let Err(e) = "10".parse::<i32>() {
        panic!("Oops {e} ")
    };
}
```
![Screen shot of error](https://user-images.githubusercontent.com/43485962/225458562-bf6e1407-5638-46b0-a205-5a60247a6e41.png)

**Comment by c-git** [comment]
This plugin has been discontinued IIRC.
